src/com/android/calendar/settings/GeneralPreferences.kt

```diff
@@ -292,8 +292,10 @@ class GeneralPreferences : PreferenceFragmentCompat(),
                 a.recreate()
             }
             KEY_PURE_BLACK_NIGHT_MODE -> {
-                Utils.sendUpdateWidgetIntent(a)
-                a.recreate()
+                if (themePref.value == "system") {
```

Comment:
```suggestion
                if (DynamicTheme.isSystemInDarkTheme(a)) {
```
Not sure if this is better, just a proposal. I tried to reproduce the flickering but didn't see it. As it is for the widget, feel free to do what you think fits best.